### DISCUSSION TITLE
Dynamically set environment (or secret) variable name

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#65828] [Actions] [answered]
### Select Topic Area

Question

### Body

I am creating a re-usable workflow that will have a `secrets: inherit` directive.

Therefore it will inherit the secrets from the invoking workflow.

I want to build a convention where some secrets will be named according to the `Dockerfile`. So for example if the dockerfile is named `Dockerfile` the convention is that these secrets will be stored in a secret variable named `Dockerfile_BUILD_SECRETS`

However the following shell step fails

```
      - name: get build secrets
        shell: bash
        run: |
          SECRETS_NAME="${{ inputs.dockerfile }}_BUILD_SECRETS"
          BUILD_SECRETS="${{ secrets.${SECRETS_NAME} }}"
          echo "BUILD_SECRETS=$BUILD_SECRETS" >> $GITHUB_ENV
```

with the error:

```
Unexpected symbol: '${SECRETS_NAME}'. Located at position 9 within expression: secrets.${SECRETS_NAME}
```

Is there a way around this?

**Comment by ChristopherHX** [comment, marked as answer]
For example by not mixing bash and GitHub Actions syntax:

```yaml
on:
  workflow_call:
    inputs:
      dockerfile:
        type: string
jobs:
  jobid:
    runs-on: ubuntu-latest
    env:
      BUILD_SECRETS: ${{ secrets[format('{0}_BUILD_SECRETS', inputs.dockerfile)] }}
    steps:
    - run: echo "$BUILD_SECRETS"
